zerver/lib/test_classes.py

```diff
@@ -135,6 +135,16 @@ def set_http_host(self, kwargs: Dict[str, Any]) -> None:
         elif 'HTTP_HOST' not in kwargs:
             kwargs['HTTP_HOST'] = Realm.host_for_subdomain(self.DEFAULT_SUBDOMAIN)
 
+        if kwargs.get('omit_http_agent'):
+            assert 'HTTP_USER_AGENT' not in kwargs
+            del kwargs['omit_http_agent']
+        elif 'HTTP_USER_AGENT' not in kwargs:
+            # Unless a caller tells us specifically to
+            # omit the user agent, we just default it to
+            # 'website', so that we avoid creating new
+            # Client records on each test.
+            kwargs['HTTP_USER_AGENT'] = 'website'
```

Comment:
I don't understand this commit; doesn't `get_client_name` in `zerver/decorator.py` default to `website` already anyway?

Reply:
I was seeing it return "Unspecified" in the typing test, and there is something clearly broken here, since adding in `HTTP_USER_AGENT` reduced the query counts for nearly every place we counted queries surrounding an entire post/get.

You should look at `get_client_name` more closely--perhaps the problem here is that `is_browser_view` is `False` when you expected it to be `True`?  Let's try to debug this together on chat.

Reply:
The caller here is `validate_api_key`, which doesn't set `is_browser_view`.  And we often test using API keys now, which I think might have been a cleanup from the last couple years or so?

Reply:
Well, if it's `api_get`, not `client_get`, then indeed is an API request, not a website request.

I think what we should be doing is setting `HTTP_USER_AGENT` to either a random Zulip desktop app `User-Agent` string or the python-zulip-api `User-Agent`, depending on whether it's a `client_get` or `api_get` that's being used for the HTTP request.  Otherwise we're replacing the weird behavior of having no User-Agent with a super weird/malformed one.

Reply:
I opened a separate PR to replace this commit: https://github.com/zulip/zulip/pull/14020